sound/soc/sof/ipc.c

```diff
@@ -296,10 +297,17 @@ static void ipc_tx_next_msg(struct work_struct *work)
 
 	/* send first message in TX list */
 	msg = list_first_entry(&ipc->tx_list, struct snd_sof_ipc_msg, list);
-	list_move(&msg->list, &ipc->reply_list);
-	snd_sof_dsp_send_msg(sdev, msg);
+	ret = snd_sof_dsp_send_msg(sdev, msg);
+	if (ret < 0) {
+		dev_err(sdev->dev, "error: ipc tx failed with error %d\n", ret);
```

Comment:
maybe dev_err_ratelimited?

Reply:
also maybe add a comment that because the ipc is not moved it'll be retried until it succeeds or timeouts?

Reply:
@plbossart fixed now.